client/state/terms/reducer.js

```diff
@@ -50,6 +55,34 @@ export function requesting( state = {}, action ) {
 }
 
 /**
+ * Returns the updated term query state after an action has been dispatched.
+ * The state reflects a mapping of serialized query key to an array of term IDs
+ * for the query, if a query response was successfully received.
+ *
+ * @param  {Object} state  Current state
+ * @param  {Object} action Action payload
+ * @return {Object}        Updated state
+ */
+export function queries( state = {}, action ) {
+	switch ( action.type ) {
+		case TERMS_REQUEST_SUCCESS:
+			const serializedQuery = getSerializedTaxonomyTermsQuery( action.query, action.taxonomy, action.siteId );
+			return Object.assign( {}, state, {
+				[ serializedQuery ]: action.terms.map( ( term ) => term.ID )
```

Comment:
`TERMS_REQUEST_SUCCESS` doesn't include `terms` in the `action` object. I suppose you meant `TERMS_RECEIVE`?